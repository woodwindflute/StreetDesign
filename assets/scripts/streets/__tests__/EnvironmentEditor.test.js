/* eslint-env jest */
import React from 'react'
import { render } from '../../../../test/helpers/render'
import EnvironmentEditor from '../EnvironmentEditor'

describe('EnvironmentEditor', () => {
  const initialState = {
    street: {
      environment: null
    },
    ui: {
      toolboxVisible: true
    }
  }

  it('renders', () => {
    const { asFragment } = render(<EnvironmentEditor />, {
      initialState
    })
    expect(asFragment()).toMatchSnapshot()
  })

  it.todo('selects an environment')
  it.todo('closes when close button is clicked')
})
