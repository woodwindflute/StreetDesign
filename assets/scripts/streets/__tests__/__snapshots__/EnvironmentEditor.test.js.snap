// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvironmentEditor renders 1`] = `
<DocumentFragment>
  <div
    class="environment-editor environment-editor-container-outer react-draggable"
    style="transform: translate(0px,0px);"
  >
    <div
      class="environment-editor-container-inner"
      style="opacity: 0; transform: scale(0.75); pointer-events: none;"
    >
      <header>
        <img
          alt="Sun behind rain cloud"
          draggable="false"
          src="test-file-stub"
        />
        <h3>
          Environment
        </h3>
        <button
          class="close"
          title="Dismiss"
        >
          <svg
            class="octicon"
          />
        </button>
      </header>
      <div
        class="environment-editor-content"
      >
        <div
          class="environment-select-grid"
        >
          <div
            class="environment-select environment-active"
            style="background: rgb(169, 204, 219);"
            title="Day"
          />
          <div
            class="environment-select"
            title="Twilight"
          />
          <div
            class="environment-select"
            title="Dusk"
          />
          <div
            class="environment-select"
            title="Sunrise"
          />
          <div
            class="environment-select"
            title="Night"
          >
            <img
              alt=""
              draggable="false"
              style="width: 100%; height: 100%; pointer-events: none; user-select: none;"
            />
          </div>
          <div
            class="environment-select"
            style="background: rgb(226, 240, 242);"
            title="Foggy"
          />
          <div
            class="environment-select"
            style="background: rgb(20, 20, 20);"
            title="Holodeck"
          >
            <img
              alt=""
              draggable="false"
              style="width: 100%; height: 100%; pointer-events: none; user-select: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
